server/neptune/workflows/internal/activities/main.go

```diff
@@ -37,13 +52,46 @@ type Terraform struct {
 	*cleanupActivities
 }
 
-func NewTerraform(serverURL *url.URL) *Terraform {
+func NewTerraform(config neptune.TerraformConfig, dataDir string, scope tally.Scope, serverURL *url.URL) (*Terraform, error) {
+	binDir, err := mkSubDir(dataDir, BinDirName)
+	if err != nil {
+		return nil, err
+	}
+
+	cacheDir, err := mkSubDir(dataDir, TerraformPluginCacheDirName)
+	if err != nil {
+		return nil, err
+	}
```

Comment:
Nish originally had a comment around this. If we do all the setup in the calling server file, it'll eventually be a giant file (similar to our current server.go file) which would be very hard to follow. So, I have tried to instantiate the common dependencies in the calling server file and the rest of the terraform specific stuff here. Since the binDir and cacheDir is only used by terraform activities, I think it makes sense to instantiate it here